### PR TITLE
[Feat] #155 - edit album cover UI 구현

### PR DESCRIPTION
**Pull request by hongjunehuke**
##  작업 내용

- edit album cover UI 구현

## 스크린샷

Simulator Screen Recording - iPhone 14 - 2023-07-19 at 17 47 36

## 관련 이슈

- Resolved: #155

### CODE REVIEW

**Review by dannaward — CHANGES_REQUESTED** [review]

File: pophory-iOS/Screen/HomeAlbum/HomeAlbumView.swift [hunk 1]
```diff
@@ -100,20 +130,41 @@ final class HomeAlbumView: UIView, GettableHomeAlbumProperty {
...
+            $0.height.equalTo(6)
+            $0.width.equalTo(0)
+            $0.centerY.equalTo(statusLabel)
+            $0.leading.equalToSuperview().inset(50)
+        }
+        progressBarView.bringSubviewToFront(self)
+        
+        progressBarIcon.snp.makeConstraints {
+            $0.centerY.equalTo(progressBarView)
+            $0.trailing.equalTo(progressBarView.snp.trailing).offset(6)
```
[the remark this answers is not in the document]
Comment (hongjunehuke) [reply pophory-iOS/Screen/HomeAlbum/HomeAlbumView.swift]: [07e22c7](https://github.com/TeamPophory/pophory-iOS/pull/156/commits/07e22c76c9ebb700424c2435214bb03eae4c648d)

File: pophory-iOS/Screen/HomeAlbum/HomeAlbumView.swift [hunk 2]
```diff
@@ -17,28 +17,41 @@ protocol ImageViewDidTappedProtocol {
...
-        label.font = .h1
-        label.text = "포릿만의 네컷 앨범에\n소중한 추억을 보관해봐!"
-        label.asColor(targetString: "포릿만의 네컷 앨범", color: .pophoryPurple)
+        label.font = .head1Medium
         label.numberOfLines = 2
+        
+        let attributedText = NSMutableAttributedString(string: "포포리 앨범에\n소중한 추억을 보관해 봐!")
+        attributedText.addAttributes([.font: UIFont.head1Bold], range: NSRange(location: 0, length: 6))
+        attributedText.addAttribute(.foregroundColor, value: UIColor.pophoryPurple, range: NSRange(location: 0, length: 6))
+        label.attributedText = attributedText
```
[the remark this answers is not in the document]
Comment (hongjunehuke) [reply pophory-iOS/Screen/HomeAlbum/HomeAlbumView.swift]: [07e22c7](https://github.com/TeamPophory/pophory-iOS/pull/156/commits/07e22c76c9ebb700424c2435214bb03eae4c648d)

File: pophory-iOS/Screen/EditAlbumCover/ViewController/EditAlbumViewController.swift [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+        case 0:
+            cell.configCell(albumCoverImage: ImageLiterals.albumCover1)
+        case 1:
+            cell.configCell(albumCoverImage: ImageLiterals.albumCover2)
+        case 2:
+            cell.configCell(albumCoverImage: ImageLiterals.albumCover3)
+        case 3:
+            cell.configCell(albumCoverImage: ImageLiterals.albumCover4)
+        default:
+            return cell
```
[the remark this answers is not in the document]
Comment (hongjunehuke) [reply pophory-iOS/Screen/EditAlbumCover/ViewController/EditAlbumViewController.swift]: [07e22c7](https://github.com/TeamPophory/pophory-iOS/pull/156/commits/07e22c76c9ebb700424c2435214bb03eae4c648d)

File: pophory-iOS/Screen/HomeAlbum/HomeAlbumViewController.swift [hunk 4]
```diff
@@ -19,8 +20,26 @@ final class HomeAlbumViewController: BaseViewController {
...
+                    let progressValue = Int(round(progressBackgroundViewWidth * (Double(photoCount) / 15.0)))
+                    homeAlbumView.updateProgressBarWidth(updateWidth: progressValue)
+                    let isAlbumFull = (photoCount == 15) ? true : false
+                    homeAlbumView.updateProgressBarIcon(isAlbumFull: isAlbumFull)
+                    
+                    // MARK: - alert
+                    
+                    if isAlbumFull {
+                        showPopup(
+                            image : ImageLiterals.img_albumfull,
```
[the remark this answers is not in the document]
Comment (hongjunehuke) [reply pophory-iOS/Screen/HomeAlbum/HomeAlbumViewController.swift]: [07e22c7](https://github.com/TeamPophory/pophory-iOS/pull/156/commits/07e22c76c9ebb700424c2435214bb03eae4c648d)

File: pophory-iOS/Screen/EditAlbumCover/View/EditAlbumView.swift [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+            $0.top.equalTo(lineView.snp.bottom).offset(52)
+            $0.leading.equalTo(albumCoverProfile2.snp.trailing).offset(18)
+            $0.size.equalTo(50)
+        }
+        
+        albumCoverProfile4.snp.makeConstraints {
+            $0.top.equalTo(lineView.snp.bottom).offset(52)
+            $0.leading.equalTo(albumCoverProfile3.snp.trailing).offset(18)
+            $0.size.equalTo(50)
+        }
```
[the remark this answers is not in the document]
Comment (hongjunehuke) [reply pophory-iOS/Screen/EditAlbumCover/View/EditAlbumView.swift]: [07e22c7](https://github.com/TeamPophory/pophory-iOS/pull/156/commits/07e22c76c9ebb700424c2435214bb03eae4c648d)

File: pophory-iOS/Screen/EditAlbumCover/View/EditAlbumView.swift [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+    }()
+    private let albumCoverProfile3: UIButton = {
+        let button = UIButton()
+        button.setImage(ImageLiterals.albumCoverProfile3, for: .normal)
+        return button
+    }()
+    private let albumCoverProfile4: UIButton = {
+        let button = UIButton()
+        button.setImage(ImageLiterals.albumCoverProfile4, for: .normal)
+        return button
```
[the remark this answers is not in the document]
Comment (hongjunehuke) [reply pophory-iOS/Screen/EditAlbumCover/View/EditAlbumView.swift]: [07e22c7](https://github.com/TeamPophory/pophory-iOS/pull/156/commits/07e22c76c9ebb700424c2435214bb03eae4c648d)

File: pophory-iOS/Screen/HomeAlbum/HomeAlbumView.swift [hunk 7]
```diff
@@ -17,28 +17,41 @@ protocol ImageViewDidTappedProtocol {
...
+    var homeAlbumViewButtonTappedDelegate: HomeAlbumViewButtonTappedProtocol?
     
     var statusLabelText: String {
         get {
             return privateStatusLabelText
         }
         set {
-            privateStatusLabelText = newValue
-            self.statusLabel.text = privateStatusLabelText + "/30"
+            privateStatusLabelText = newValue + "/15"
```
[the remark this answers is not in the document]
Comment (hongjunehuke) [reply pophory-iOS/Screen/HomeAlbum/HomeAlbumView.swift]: [07e22c7](https://github.com/TeamPophory/pophory-iOS/pull/156/commits/07e22c76c9ebb700424c2435214bb03eae4c648d)

File: pophory-iOS/Screen/EditAlbumCover/ViewController/EditAlbumViewController.swift [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+}
+
+extension EditAlbumViewController: Navigatable {
+    var navigationBarTitleText: String? { return "앨범 테마" }
+}
+
+extension EditAlbumViewController: UICollectionViewDataSource {
+  
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return 4
```
[the remark this answers is not in the document]
Comment (hongjunehuke) [reply pophory-iOS/Screen/EditAlbumCover/ViewController/EditAlbumViewController.swift]: [07e22c7](https://github.com/TeamPophory/pophory-iOS/pull/156/commits/07e22c76c9ebb700424c2435214bb03eae4c648d)

File: pophory-iOS/Screen/EditAlbumCover/View/EditAlbumView.swift
[the remark this answers is not in the document]
Comment (dannaward) [reply pophory-iOS/Screen/EditAlbumCover/View/EditAlbumView.swift:1]: 따로 이슈 파서 진행하는 걸로!

**Review by dannaward — APPROVED** [review]

File: pophory-iOS/Screen/EditAlbumCover/View/EditAlbumView.swift
[the remark this answers is not in the document]
Comment (hongjunehuke) [reply pophory-iOS/Screen/EditAlbumCover/View/EditAlbumView.swift:1]: https://github.com/TeamPophory/pophory-iOS/pull/167

**Pull request merged**